### ISSUE TITLE
State Description noct transfered from things.xml to Item

### ISSUE THREAD

**Issue by volkmarnissen** [openhab/openhab-core#2815]
I created the following channel-Type in OH-INF/thing/things.xml for a binding:
	<channel-type id="chargingModeType">
		<item-type>String</item-type>
		<label>Charging Mode</label>
		<description>The current charging mode of the device (One of Boost, Fast, Eco, Eco+ or Stop)</description>
		<state readOnly="false">
			**<options>
				<option value="0">Boost</option>
				<option value="1">Fast</option>
				<option value="2">Eco</option>
				<option value="3">Eco+</option>
				<option value="4">Stop</option>
			</options>**
		</state>
	</channel-type>

If I create an Equipment from Thing in the model, the state description  is not available in the item and it doesn't have the options.

This happens in OH 3.3 main branch.

**Comment by J-N-K** [comment]
Can you please show the full binding code?

**Comment by volkmarnissen** [comment]
It's at
https://github.com/volkmarnissen/openhab-addons/tree/myenergi_binding1/bundles/org.openhab.binding.myenergi

**Comment by J-N-K** [comment]
Works for me. I cloned your repository, compiled the binding and put it into the addons-folder. Then I created a bridge and a `zappi` thing (of course nothing came online). I linked a new `String` item `Zappi_ChargingMode` to the `device#chargingMode` channel. The state was `NULL` at that time. I then used the console to update the item state: `openhab:update Zappi_ChargingMode 1` and the UI changed to `Fast`, which I believe is correct.

Did you add the state description to the binding after the thing was created?

**Comment by volkmarnissen** [comment]
no, I added the state description in things.xml before the thing was created.

May be I have the wrong understanding how it should work.
I can still reproduce it as follows:
1. Create the Zappi Thing
2. In the Model: Add Equipment from Thing for Thing: Zappi, Channel "Charging Mode".
3. Save it.
4. Look at the newly created items Metadata
I expected to see a State Description with filled options field.
However, there is no State Description generated.

**Comment by J-N-K** [comment]
No, the state description is not copied to the item metadata. It is applied if no item metadata overrides it.

**Comment by Rossko57** [comment]
Comment - I believe using REST API  to inspect the Item allows you to see the _effective_ properties, including any phantom options from bindings.

**Comment by volkmarnissen** [comment]
I am a newbie, and I didn't use the REST API as of now. May be this explains my concerns.

How is it from the end user perspective?
An end user will not use REST API.

I as an end user want to configure a binding, select some channels from the list, then generate items out of it.
I expect the items to work as well as possible.
Currently, the items don't "know" their state descriptions. So, I have to modify them and duplicate the values from the xml file.

If my "end user" expectations are wrong. Just close the issue. 
Now, that I understood how it works, it's no longer an issue for me personally.

**Comment by Rossko57** [comment]
> How is it from the end user perspective?
> An end user will not use REST API

No - but the UI widgets do use REST API.  Thewhole purpose of the Item state/command options is for the widgets to use in user interaction, without any intervention.
So as a designer you can use REST API to inspect what your design makes available to widgets via a linked Item.

**Comment by volkmarnissen** [comment]
>So as a binding developer you can use REST API to inspect what your design makes available to widgets via a linked Item.
How can I do it?
I found https://www.openhab.org/docs/configuration/restdocs.html. It deals with accessing the REST API from outside.
How can I access the REST API from the binding?
Are there examples?

**Comment by Rossko57** [comment]
I don't know what you're trying to do.
To create Item options, use XML.

If you want to see if that worked, you (sat at your keyboard) can use REST API.

**Comment by volkmarnissen** [comment]
I assume, here is some misunderstanding which is not so relevant and the issue is not a problem for me. I'll close it.
